### DISCUSSION TITLE
Vision for Transportr

### DISCUSSION THREAD

**Discussion by Echolon** [grote/Transportr#858] [Ideas]
Dear Community,

I think the feedback on my issue here is tremendous good and I am happy so many are still in favor of this!

Hence, I thought we could discuss also a bit of a _vision_ for this app. I know this sounds a bit to much for where we are, but I think it is a nice motivational instrument. And even if we just come up with goals / targets it is something where we can all head for in the end and also motivate others to join this effort. That could at least help to set the scene of what we can work on next.

What do you think?

**Comment by Echolon** [comment]
One part of the vision could be to set a coverage target. I think the core value of the app is coverage. So let's increase (once we have wramped up everything) the coverage and call for more contributions in this regard. I think the biggest value the app adds is access to as much as networks as possible. So if we increase this number and also advertise this a little bit then we can quickly increase the impact.

To argue what we could do first would be to start with networks with the most people using that network (number of people in the city for example). I started with a list of open request in the issue list. Another interesting issue could be to improve here: https://github.com/grote/Transportr/issues/655 fixing existing networks would be another attempt.

Sample:
| Network | Covered | Issues | Potential Users | Comment |
|---------|---------|----------|--------|--------|
| Dehli, India   |  False       |     https://github.com/grote/Transportr/issues/461     |    31.870.000    |        |
| Melbourne, Australia |  True       |    https://github.com/grote/Transportr/issues/767      |     4.590.000     |        |
| Auckland, New Zealand  | False   | https://github.com/grote/Transportr/issues/748       |     509.625  |      |
| Windhoek, Namibia  | False    | https://github.com/grote/Transportr/issues/743     |   486.000   |        |
| VVV, VOR and STV, Austria    |  True       |    https://github.com/grote/Transportr/issues/817      |    ?    |        |
| UK    |  True       |      https://github.com/grote/Transportr/issues/886    |    ?    |        |

![Screenshot_20230522_213150](https://github.com/grote/Transportr/assets/29168011/2d0758a1-9d46-433f-ae86-1601cba016c7)
(https://transportr.app/)

**Reply by ialokim** [reply]
Also, the coverage map is currently probably outdated. Might be good to double-check at some point.

**Reply by Altonss** [reply]
Has this map been generated manually?

**Reply by ialokim** [reply]
I'd guess so, yes: https://github.com/grote/Transportr-website/blob/master/layouts/partials/coverage-map.html

**Reply by Echolon** [reply]
https://github.com/grote/Transportr-website/issues/13

**Reply by ghost** [reply]
I could work on AZ (USA) and surrounding states if I'll be able to get the handle of it (might take some learning but from what I've seen it's not terribly complicated). GTFS data is already avaible in various states so it's possible to do. Magic Earth has a list of their transit coverage right [here](https://www.magicearth.com/feature-availablity/#transit).

Also, Poland has a well functioning transit app with pretty much all major cities (even the smaller ones offer GTFS iirc), but it isn't FOSS as many would like it to be, so there could be a contributions made there as well. 😃

Talking mainly about buses/metro/light rail & just throwing the idea out there, so if I'm missing something or it isn't as simple as it looks, I apologize and please correct me! ❤️

**Reply by guushoekman** [reply]
One inaccuracy of the map that I know of is that it says the Netherlands is covered.

The app currently shows two sources: NS (Dutch Railways, government owned - [ns.nl](https://www.ns.nl/)) and 9292 (private company - [9292.nl](https://9292.nl/en/)).

Neither works, so at this point there is no coverage at all for the Netherlands. The app does list both of these but then any attempt at getting route suggestions will fail.

I attempted to fix it and add coverage again (https://github.com/schildbach/public-transport-enabler/pull/510/) but was told Navitia based providers were going to be removed.

That's a shame but the NS has an API that I believe is free to use (for non-commercial purposes). Even though it's Dutch Railways their API contains all modes of public transport for the whole country so implementing that would provide coverage for the whole country.

https://apiportal.ns.nl/api-details#api=reisinformatie-api (you might not see anything there because it requires a login).

**Reply by Echolon** [reply]
@guushoekman maybe you can formulate this in an issue?

**Comment by genofire** [comment]
Get ib F-Droid again, for years: see #826 and #835

**Comment by genofire** [comment]
bookmark trips with times (easier lookup and refresh if during trip a trains gone be late)

**Reply by ialokim** [reply]
This is tracked in #171. Please continue discussion/upvote there.

**Comment by genofire** [comment]
a possibility to store the deutschland-tickets:
https://deutschlandticket.de/

or other codes / qr scans or screenshots of (qr)codes

**Reply by Altonss** [reply]
I don't think this is a good idea, at least it is shouldn't be high priority. I prefer to keep things simple, stupid (as we already struggle for maintenance). Other apps to it very well like Catima or PassAndroid.

**Reply by Echolon** [reply]
genofire maybe use a tool like this: https://apps.kde.org/itinerary/

**Comment by ghost** [comment]
If we get the app into a usable state, maybe we could contact some of the more known people in the FOSS community on YouTube or such, so more people know about the existance of an open-source alternative for public transit? A good idea might be to ask people like Louis Rossman on YT to cover the app and its capabilities when there's suport for Texas and NYC.

**Comment by Mannshoch** [comment]
Pleas remove the manual coose of Country and railway.

- [ ] Auto choose the source by choosed station. 
- [ ] Use Automatic a fallback if a source not work for a station
- [ ] push an error message if a source not work on a station to transportr

**Reply by Altonss** [reply]
I understand this request from the user perspective, but the reality of things is much more complicated.
For the search we rely on chosen network API. So this means that for now we cannot auto-choose the source.
Also, for some trips several networks might be available, and it's difficult to select which one to use.

**Reply by Mannshoch** [reply]
I think thats what defines a Vision. A definition of best cases that you have in your mind while you work. You do not have to have a solution in Mind while you write a Vision.
It has not to be in the next release.
If so, it's not a vision, it would be a projectplan.

**Reply by Mannshoch** [reply]
May a question, could you search a station by the name it has on openstreetmap?
My exprience is, the station name is the same like used by the networks (on frontend). The station name give you the location and therefor the possible networks. 
One problem I experience if I use transportr is, that while searching a station, the station  from the "FROM:" field has no effect on the "TO:" field. Many times - on city trips - I really have to search with the full name of the station because names from far away are shown first. I hope that would be solvable by using OSM.

As a vision, you could add a Login for openstreetmap and if trainstations have meta information for used ids by api, you may could add infromation to OSM for faster match trainstation and network.
An even greater vision could be to let user add information to help match missing platform on a trainstation.

**Comment by Mannshoch** [comment]
I would like to support https://github.com/grote/Transportr/issues/579 for Landscape 

and 
 A Tilling swipe surface for daily users.
https://github.com/grote/Transportr/issues/813

**Reply by Altonss** [reply]
What would be the use-case for landscape mode?

**Reply by Mannshoch** [reply]
If I think about it, I assume the point is a fast switch of view. In my opinion, a menu to change appearence may better.

**Reply by Altonss** [reply]
I never used Transportr in landscape mode. In which context/use-case do you need/use the Transportr in landscape mode?

**Comment by Mannshoch** [comment]
I'm a User of this app.
I user public transport daily. So my propose for a vision would be an app that is my companion on daily travels. Between a limited set of same stations and sometimes to random stations.
So I also need a way to see delays.
https://github.com/grote/Transportr/issues/903

**Reply by Altonss** [reply]
Delays can already be shown in Transportr. What network are you using?

**Comment by Echolon** [comment]
Hi all,

I just updated my vision with a small suggestion: https://github.com/grote/Transportr/discussions/858#discussioncomment-5971160

With a certain goal, we may even be able to call for donations.

**Comment by Zack-83** [comment]
Hallo, even if I am joining the discussion quite late, I strongly support your initiative!  
In my opiniom, the strengths of Transportr are the map and specially the function to find stops near a point on the map, which is life-saving when I travel in a new region.  
Did you ever think about a cooperation with the other transport app, Öffi?  
Some possible extensions:  
- Let a search start after clicking a "start" button (not all users search a connection starting immediately)
- Allow searching connections beyond the borders of a transport region
- Add transport network maps
Off-topic: can anyone recommend a good software to create schematic network maps, possibly scalable?

**Comment by OpensourceGhost** [comment]
Any updates on real time integration in Canada, US, and UK transport services seems to be only a few for US and Canada and not anything on East and Wes-tcoast what would be the best way to help? Is Transitious the best alternative at this time?

**Reply by Echolon** [reply]
@OpensourceGhost see the recent release

**Comment by Echolon** [comment]
@woheller69 maybe this discussion is interesting, too.

And thanks for your efforts! If you are interested to continue, we could maybe discuss and plan the way forward together with @Altonss who has been contributing recently. Having some level of review would be great, too.

Another matter could also be what we could change so this becomes more sustainable. And it would be great if we could revive promoting via e.g. mastodon and what not. I could help here.

**Reply by woheller69** [reply]
v2.2.5 coming soon, where I fixed some bugs and added the clear all trips feature.
With that I - personally - will be happy for a while.
Unfortunatly this app is written in a way which I do not like very much - using viewmodels and Kotlin ...
So I won't be able to support major redesigns.

**Reply by Echolon** [reply]
That's fine and I understand.

**Reply by woheller69** [reply]
Btw, a beta of 2.2.5 is here for testing:
https://github.com/woheller69/Transportr/releases/tag/V2.2.5

**Reply by Echolon** [reply]
> v2.2.5 coming soon, where I fixed some bugs and added the clear all trips feature. With that I - personally - will be happy for a while. Unfortunatly this app is written in a way which I do not like very much - using viewmodels and Kotlin ... So I won't be able to support major redesigns.

This does not influence the dislike around viewmodels, right? https://github.com/grote/Transportr/pull/927

**Reply by woheller69** [reply]
I see no need to change to MD3. The app works great as it is. And for people who do not need any features but desperately want MD3 there is TransportYou...

**Reply by woheller69** [reply]
not to forget - once the change is done, Google will have invented a new theme which again requires a complete redesign
